### PR TITLE
[SocialSharing] Dynamic calculation of reward amount based on social stats

### PR DESCRIPTION
**Pull request by franckc**
### Description:

Updated the rules engine so that the new SocialShare rule can calculate reward amount based on the user's social stats. There are 2 scenarios:
 - before the user completed the sharing action, we use the social stats from the user's identity. The social stats will be written to the identity table by the event-listener, when a user publish their profile. The source of the data is the attestation table that gets written by the bridge when a user does an attestation.
 - after the user completed the sharing action, we use the social stats from the GrowthEvent row that was logged by the bridge server when the post was verified.

Note: the calculation function does not implement fully yet [the spec from micah](https://docs.google.com/spreadsheets/d/1fJ1BVNSDnWFqKh_L0zwxhhklGLalxLbGS1Ss3B1xfvg/edit?ts=5d3f7715#gid=0). More specifically, it does not check:
 - whether or not account is verified
 - date of last tweet.
Once we confirm for sure we can pull those stats, I'll send a separate PR to update the function.

### Checklist:

- [X] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

**Comment by franckc** [comment]
This is now ready for review.

**Comment by sparrowDom** [comment]
1 comment inline otherwise lgtm

**Comment by franckc** [comment]
@sparrowDom Sorry, I could not find your comment ? did you forget to publish ? I'm going to merge this in so that we can start integrating the pieces but I'm happy to address your feedback as a separate PR. Thanks :)

File: infra/growth/src/resources/rules.js [hunk 1]
```diff
@@ -731,7 +752,118 @@ class SocialShareRule extends SingleEventRule {
...
   /**
-   * Hashes content for verification of the user's post purposes.
+   * Calculates the personalized reward amount based on number of followers
+   * and age of the user's twitter account.
+   *
+   * @param {numFollowers:number, accountAge:number}
+   * @returns {<number>}
+   * @private
+   */
+  _calcTwitterReward(stats) {
```
[the remark this answers is not in the document]
Comment (franckc) [reply infra/growth/src/resources/rules.js]: @sparrowDom Very good point ! Either we can store the constant in the config or alternatively the name of the function ? Let's see what formula product wants to use for next month's campaign...

[the remark this answers is not in the document]
Comment (sparrowDom) [reply infra/growth/src/resources/rules.js]: nice, sounds good

**Pull request merged**